### DISCUSSION TITLE
How safely change a field type?

### DISCUSSION THREAD

**Discussion by Danbka** [webonyx/graphql-php#1586] [General]
Hey,

When we deprecate the whole field in the schema the flow is:
1. Mark the field deprecated
2. Add new field (if needed)
3. When we are sure nobody uses the deprecated field - we can remove it.

But what if I need to change the field type (int to float e.g.)? The main problem for me, if we follow the flow above, is inventing a new name for the new field. Any ideas how to manage it properly?

**Comment by spawnia** [comment]
I generally see no other option then the deprecation flow you described.

If you know all clients, you could change them to accept both types, change the type, then settle on the new one.
